src/DotLiquid/StandardFilters.cs

```diff
@@ -166,15 +166,18 @@ public static string H(string input)
         /// <param name="length"></param>
         /// <param name="truncateString"></param>
         /// <returns></returns>
-        public static string Truncate(string input, int length = 50, string truncateString = "...")
+        public static string Truncate(string input, long length = 50, string truncateString = "...")
         {
             if (string.IsNullOrEmpty(input))
                 return input;
 
-            int l = length - truncateString.Length;
+            if (length < 0 || length > int.MaxValue)

```

Comment:
I see how check for negative values can be an improvement. However it is also a breaking change as calls which here and below for truncate words returned just the ellipsis would now throw an argument exception or liquid error.

Reply:
OK, I'm happy to remove this check if its already handled elsewhere - if you want me to continue? What's your plan? Are you going to update and merge your pull request, and then add this one on top?

Reply:
Glad you are here. I've tried to take all your new test cases and some of your code improvement in my PR. There may not be a need for your PR on top, apologies to the 'take over'. Let me know if I missed sometime.

Reply:
That's fine - you can take over. I'll just close this if that's OK?

Reply:
Picked two good coding ideas and adopted them, plus adding pattern matching on top.

Reply:
Negative values for truncate and truncate words currently lead to just the truncate string to be returned. As it was not always obvious from reading the code, I've done small refactoring in my PR to make this explicit (and preserve current behavior).
Beside truncate and truncate-words my February PR merge broke a number of other scenarios so I restored, fixed and added systematically all the unit tests that were touched back then.
I hope that my PR now is including all of the corrections that you need, as well as the good ideas you added.